src/module_6/repositories/cart.ts

```diff
@@ -1,40 +1,27 @@
-import { CartsEntity } from '../dataBase/carts'
 import {
     fetchCartAndTotalPrice,
-    getOrderData,
 } from '../utils';
-import { DI } from '../../../app';
-import Cart from '../../module_7/entities/cart';
 import { getProductById } from './product';
-import Order from '../../module_7/entities/order';
-import { getUser } from './users';
-
-export interface Product {
-    productId: string
-    count: number
-}
-
-export interface CartTemplate {
-    cart: Pick<CartsEntity, 'id' | 'items'>
-    total: number
-}
-
-const getActiveCart = async (userId: string): Promise<Cart | null> => {
-    const user = await getUser(userId);
-    const cart = await DI.cart.findOne({ user, isDeleted: false });
+import {
+    CartTemplate,
+    ProductData,
+    CartModelAndMethods
+} from '../../module_7/entities/types';
+import Cart from '../../module_7/entities/cart';
 
-    return cart;
+const getActiveCart = async (userId: string): Promise<CartModelAndMethods | null> => {
+    return await Cart.findOne({ user: userId, isDeleted: false }).exec();
 }
 
-export const createCart = async (userId: string): Promise<Cart | null> => {
-    const user = await getUser(userId);
-    if (user) {
-        const newCart = new Cart(user);
-        await DI.em.persistAndFlush(newCart);
+export const createCart = async (userId: string): Promise<CartModelAndMethods | null> => {
+    const newCart = new Cart({
+        user: userId,
+        items: [],
+    });
```

Comment:
Better move all business logic to services and use repositories just for db calls